documentation/src/docs/wire-protocol.md

```diff
@@ -7,106 +7,103 @@ You may obtain a copy of the License at
 
     http://www.apache.org/licenses/LICENSE-2.0
 -->
-# Streaming Service Wire Protocol
+# Pravega Streaming Service Wire Protocol
 
-This page describes the proposed Wire Protocol for the Streaming Service. See parent page for description of the service as a whole.
+This page describes the proposed Wire Protocol for the Streaming Service. See [Pravega Concepts](http://pravega.io/docs/latest/pravega-concepts) for more information.
 
-Protocol
-========
+# Protocol
 
-Data is sent over the wire in self-contained "messages" that are either "requests" (messages from the client to the server) or "replies" (which are in response to a request and go back to the client).
+Data is sent over the wire in self-contained "messages" that are either "requests" (messages sent from the client to the server) or "replies" (responses sent from the server to the client).
 
-All requests and replies have an 8 byte header with two fields, (All data is written in BigEndian format):
-1.  Message Type - An integer (4 bytes) identifies the message type and this determines what fields will follow. (Note the protocol can be extended by adding new types)
-2.  Length - Unsigned integer 4 bytes (Messages should be &lt; 2^24, but the upper bits remain zero). How many bytes of data from this point forward are part of this message. (Possibly zero, indicating there is no data)
+All the requests and replies have 8 byte headers with two fields (all data is written in **BigEndian** format).
 
-The remainder of the fields are specific to the type of message. A few important messages are listed below.
+- **Message Type**: An Integer (4 bytes) identifies the message type and determines the subsequent fields. (Note that the protocol can be extended by adding new types.)
+- **Length**:  An Integer (4 bytes) (Messages should be less than 2<sup>24</sup>, but the upper bits remain zero). Payload size of the message (possibly zero, indicating there is no data). The remainder of the fields are specific to the type of the message. A few important messages are listed below.
 
-General
--------
+# General
 
-### Partial Message - Request/Reply
+## KeepAlive - Request/Reply
 
-1.  Begin/Middle/End - Enum (1 byte)
-2.  Data
-A partial message is one that was broken up when being sent over the wire. (For any reason). The whole message is reconstructed by reading the partial messages in sequence and assembling them into a whole. It is not valid to attempt to start a new partial message before completing the previous one.
+# Reading
 
-### KeepAlive - Request/Reply
+## Read Segment - Request
 
-1.  Data - uninterpreted data of the length of the message. (Usually 0 bytes)
+1. Segment to read: String (2 byte length, followed by that many bytes of Java's Modified UTF-8).
+2. Offset to read from: Long (8 bytes).
```

Comment:
@tkaitchuck Please provide description.

Reply:
It is the offset to read from. The offset in the segment. ?

Reply:
@tkaitchuck Edited